pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -169,6 +168,32 @@ func Pod(source *v1alpha1.CatalogSource, name string, image string, saName strin
 		},
 	}
 
+	// Override scheduling options if specified
+	if source.Spec.GrpcPodConfig != nil {
+		grpcPodConfig := source.Spec.GrpcPodConfig
+
+		// Override node selector
+		if grpcPodConfig.NodeSelector != nil {
+			pod.Spec.NodeSelector = make(map[string]string, len(grpcPodConfig.NodeSelector))
+			for key, value := range grpcPodConfig.NodeSelector {
+				pod.Spec.NodeSelector[key] = value
+			}
+		}
+
+		// Override priority class name
+		if grpcPodConfig.PriorityClassName != "" {
```

Comment:
@perdasilva Now that I'm reading it, I think you're right to suggest we make this a pointer -- it leaves the option open for OLM to have some default that's not `""` and for users to opt into the original.

Reply:
Any thoughts on how we should derive configuration when the `operatorframework.io/priorityclass` annotation has been specified on a CatalogSource: https://github.com/operator-framework/operator-lifecycle-manager/blob/4fc476fc520d4d595368ffc8e4c032c4d34dd796/pkg/controller/registry/reconciler/reconciler.go#L27? I'm not entirely sure if that hacky way of propagating configuration to the underlying Pod violates any API contract? It might also be entirely reasonable to point users towards these new configuration fields in the CatalogSource entirely, but it's unknown how many users were relying on that behavior.

Reply:
this change seems to be backward compatible. If I had a priorityclass set on the catalogsource and now I add a config that contains a priority class, it's still [the former that is propagated through](https://github.com/operator-framework/operator-lifecycle-manager/pull/2512/files#diff-fffdeef1fc140a5dc5dc92dda323f567a6e46fc2ecbb0b91ba907acd02bde50dL173-L175).

Reply:
@timflannagan I've added a test to ensure that the annotation takes precedence. We'll handle this in documentation as well. We should probably deprecate the annotation tho.